NPE in enum constructors if hamcrest on classpath

Since 1.4.7, if I have hamcrest in the classpath I get a `NullPointerException` thrown in the `<init>` method of enums with constructors if I try to load them using `Class.forName` when pitest does the initial test run.

```
package com.example;

import org.junit.Test;

public class NestedClassTest {
	public static enum MyEnum {
		A { }
	}

	@Test
	public void test() throws ClassNotFoundException {
		Class.forName("com.example.NestedClassTest$MyEnum$1");
	}
}
```

```
   [pitest] 12:06:57 PIT >> INFO : MINION : 12:06:57 PIT >> INFO : 1 tests received
   [pitest] 
   [pitest] 12:06:57 PIT >> INFO : MINION : 12:06:57 PIT >> FINE : Running 1 units
   [pitest] 
   [pitest] 12:06:57 PIT >> INFO : MINION : 12:06:57 PIT >> FINE : Gathering coverage for test Description [testClass=com.example.NestedClassTest, name=test(com.example.NestedClassTest)]
   [pitest] 
   [pitest] 12:06:57 PIT >> INFO : MINION : 12:06:57 PIT >> SEVERE : Description [testClass=com.example.NestedClassTest, name=test(com.example.NestedClassTest)]
   [pitest] java.lang.ExceptionInInitializerError
   [pitest] 	at java.lang.Class.forName0(Native Method)
   [pitest] 	at java.lang.Class.forName(Class.java:264)
   [pitest] 	at com.examp
   [pitest] 12:06:57 PIT >> INFO : MINION : le.NestedClassTest.test(Unknown Source)
   [pitest] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
   [pitest] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
   [pitest] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAc
   [pitest] 12:06:57 PIT >> INFO : MINION : cessorImpl.java:43)
   [pitest] 	at java.lang.reflect.Method.invoke(Method.java:498)
   [pitest] 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
   [pitest] 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
   [pitest] 	at or
   [pitest] 12:06:57 PIT >> INFO : MINION : g.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
   [pitest] 	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
   [pitest] 	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
   [pitest] 	at org.junit.runners.B
   [pitest] 12:06:57 PIT >> INFO : MINION : lockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
   [pitest] 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
   [pitest] 	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
   [pitest] 	at org.junit.runners.ParentRunner$1.schedu
   [pitest] 12:06:57 PIT >> INFO : MINION : le(ParentRunner.java:71)
   [pitest] 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
   [pitest] 	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
   [pitest] 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
   [pitest] 	at org.junit.runners.Pa
   [pitest] 12:06:57 PIT >> INFO : MINION : rentRunner.run(ParentRunner.java:363)
   [pitest] 	at org.pitest.junit.adapter.CustomRunnerExecutor.run(CustomRunnerExecutor.java:42)
   [pitest] 	at org.pitest.junit.adapter.AdaptedJUnitTestUnit.execute(AdaptedJUnitTestUnit.java:69)
   [pitest] 	at org.pitest.coverage.execute.CoverageDecora
   [pitest] 12:06:57 PIT >> INFO : MINION : tor.execute(CoverageDecorator.java:49)
   [pitest] 	at org.pitest.testapi.execute.containers.UnContainer.execute(UnContainer.java:31)
   [pitest] 	at org.pitest.testapi.execute.Pitest.executeTests(Pitest.java:57)
   [pitest] 	at org.pitest.testapi.execute.Pitest.run(Pitest.java:48)
   [pitest] 	at org.p
   [pitest] 12:06:57 PIT >> INFO : MINION : itest.coverage.execute.CoverageWorker.run(CoverageWorker.java:49)
   [pitest] 	at org.pitest.coverage.execute.CoverageMinion.main(CoverageMinion.java:90)
   [pitest] Caused by: java.lang.NullPointerException
   [pitest] 	at com.example.NestedClassTest$MyEnum$1.<init>(Unknown Source)
   [pitest] 	at com.
   [pitest] 12:06:57 PIT >> INFO : MINION : example.NestedClassTest$MyEnum.<clinit>(Unknown Source)
   [pitest] 	... 28 more
   [pitest] 
   [pitest] 12:06:57 PIT >> SEVERE : Description [testClass=com.example.NestedClassTest, name=test(com.example.NestedClassTest)] did not pass without mutation.
   [pitest] 12:06:57 PIT >> FINE : Coverage generator Minion exited ok
   [pitest] 12:06:57 PIT >> INFO : Calculated coverage in 1 seconds.
   [pitest] 12:06:57 PIT >> SEVERE : Tests failing without mutation: 
   [pitest] Description [testClass=com.example.NestedClassTest, name=test(com.example.NestedClassTest)]
   [pitest] Exception in thread "main" org.pitest.help.PitHelpError: 1 tests did not pass without mutation when calculating line coverage. Mutation testing requires a green suite.
   [pitest] See http://pitest.org for more details.
   [pitest] 	at org.pitest.coverage.execute.DefaultCoverageGenerator.verifyBuildSuitableForMutationTesting(DefaultCoverageGenerator.java:114)
   [pitest] 	at org.pitest.coverage.execute.DefaultCoverageGenerator.calculateCoverage(DefaultCoverageGenerator.java:96)
   [pitest] 	at org.pitest.coverage.execute.DefaultCoverageGenerator.calculateCoverage(DefaultCoverageGenerator.java:51)
   [pitest] 	at org.pitest.mutationtest.tooling.MutationCoverage.runReport(MutationCoverage.java:115)
   [pitest] 	at org.pitest.mutationtest.tooling.EntryPoint.execute(EntryPoint.java:121)
   [pitest] 	at org.pitest.mutationtest.tooling.EntryPoint.execute(EntryPoint.java:51)
   [pitest] 	at org.pitest.mutationtest.commandline.MutationCoverageReport.runReport(MutationCoverageReport.java:87)
   [pitest] 	at org.pitest.mutationtest.commandline.MutationCoverageReport.main(MutationCoverageReport.java:45)
```

If I run the tests using junit directly, the exception is not thrown. If I remove hamcrest from the classpath, the exception is not thrown.

I have tried with hamcrest 1.3 and hamcrest 2.2 and both cause the exception to be thrown.

I have tried with both openjdk 11.0.6 and oracle Java JDK 8 [1.8.0_251]. Both cause the exception to be thrown.

I have attached a minimal example using pitest-ant.

[pitest-npe-in-enum-ctor.zip](https://github.com/hcoles/pitest/files/4722872/pitest-npe-in-enum-ctor.zip)

Full output:
```
Buildfile: /Users/work/Projects/pitest-ant-example/build.xml

clean:
   [delete] Deleting directory /Users/work/Projects/pitest-ant-example/build

compile:
    [mkdir] Created dir: /Users/work/Projects/pitest-ant-example/build/classes
    [javac] Compiling 1 source file to /Users/work/Projects/pitest-ant-example/build/classes
    [mkdir] Created dir: /Users/work/Projects/pitest-ant-example/build/test-classes
    [javac] Compiling 1 source file to /Users/work/Projects/pitest-ant-example/build/test-classes

pit:
   [pitest] 12:06:55 PIT >> INFO : ---------------------------------------------------------------------------
   [pitest] 12:06:55 PIT >> INFO : Enabled (+) and disabled (-) features.
   [pitest] 12:06:55 PIT >> INFO : -----------------------------------------
   [pitest] 12:06:55 PIT >> INFO : +FANN           Filters mutations in classes and methods with matching annotations of class or runtime retention
   [pitest] 12:06:55 PIT >> INFO :   [annotation]    Annotation to avoid (full package name not required)
   [pitest] 12:06:55 PIT >> INFO : +FENUM          Filters mutations in enum constructors
   [pitest] 12:06:55 PIT >> INFO : +FFBLOCK        Filters mutations in code duplicated by finally block inlining
   [pitest] 12:06:55 PIT >> INFO : +FFEACH         Filters mutations in compiler generated code that implements for each loops
   [pitest] 12:06:55 PIT >> INFO : +FFLOOP         Filters any mutations to increments in for loops as they may cause timeouts
   [pitest] 12:06:55 PIT >> INFO : +FINFINC        Filters mutations to increments that may cause infinite loops
   [pitest] 12:06:55 PIT >> INFO : +FINFIT         Filters mutations that may cause infinite loops by removing calls to iterator.next
   [pitest] 12:06:55 PIT >> INFO : +FINULL         Filters mutations in compiler generated code that checks for null by calling getClass
   [pitest] 12:06:55 PIT >> INFO : +FKOTLIN        Filters out junk mutations in bytecode created by compiler for kotlin language features
   [pitest] 12:06:55 PIT >> INFO : +FLOGCALL       Filters mutations in code that makes calls to logging frameworks
   [pitest] 12:06:55 PIT >> INFO : +FMRNULL        Filters mutations in compiler generated code that inserts Objects.requireNonNull for method references
   [pitest] 12:06:55 PIT >> INFO : +FRETEQUIV      Filters return vals mutants with bytecode equivalent to the unmutated class
   [pitest] 12:06:55 PIT >> INFO : +FSEQUIVEQUALS  Filters equivalent mutations that affect only performance in short cutting equals methods
   [pitest] 12:06:55 PIT >> INFO : +FSTATI         Filters mutations in static initializers and code called only from them
   [pitest] 12:06:55 PIT >> INFO : +FSTATINIT      Filters mutations in static initializers and code called only from them
   [pitest] 12:06:55 PIT >> INFO : +FTRYWR         Filters mutations in code generated for try with resources statements
   [pitest] 12:06:55 PIT >> INFO : -CLASSLIMIT     Limits the maximum number of mutations per class
   [pitest] 12:06:55 PIT >> INFO :   [limit]         Integer value for maximum mutations to create per class
   [pitest] 12:06:55 PIT >> INFO : -EXPORT         Exports mutants bytecode and other details to disk
   [pitest] 12:06:55 PIT >> INFO : ---------------------------------------------------------------------------
   [pitest] 12:06:56 PIT >> FINE : Running report with ReportOptions [targetClasses=[com.example.*], excludedMethods=[], excludedClasses=[], excludedTestClasses=[], codePaths=[], reportDir=pitReports, historyInputLocation=null, historyOutputLocation=null, sourceDirs=[src], classPathElements=[/Users/work/Projects/pitest-ant-example/lib/pitest-1.5.2.jar, /Users/work/Projects/pitest-ant-example/build/classes, /Users/work/Projects/pitest-ant-example/build/test-classes, /Users/work/Projects/pitest-ant-example/lib/junit-4.12.jar, /Users/work/Projects/pitest-ant-example/lib/hamcrest-2.2.jar], mutators=[], features=[], dependencyAnalysisMaxDistance=-1, jvmArgs=[-Djava.awt.headless=true], numberOfThreads=1, timeoutFactor=1.25, timeoutConstant=4000, targetTests=[^com\.example\..*$], loggingClasses=[], maxMutationsPerClass=0, verbose=true, failWhenNoMutations=true, outputs=[HTML], groupConfig=TestGroupConfig [excludedGroups=[], includedGroups=[]], fullMutationMatrix=false, mutationUnitSize=0, shouldCreateTimestampedReports=true, detectInlinedCode=false, exportLineCoverage=false, mutationThreshold=0, coverageThreshold=0, mutationEngine=gregor, javaExecutable=null, includeLaunchClasspath=false, properties={}, maxSurvivors=-1, excludedRunners=[], includedTestMethods=[], testPlugin=junit, useClasspathJar=false, skipFailingTests=false]
   [pitest] 12:06:56 PIT >> FINE : System class path is /Users/work/Projects/pitest-ant-example/lib/junit-4.12.jar:/Users/work/Projects/pitest-ant-example/lib/xstream-1.4.8.jar:/Users/work/Projects/pitest-ant-example/lib/xmlpull-1.1.3.1.jar:/Users/work/Projects/pitest-ant-example/lib/pitest-1.5.2.jar:/Users/work/Projects/pitest-ant-example/lib/pitest-entry-1.5.2.jar:/Users/work/Projects/pitest-ant-example/lib/pitest-ant-1.5.2.jar
   [pitest] 12:06:56 PIT >> FINE : Maximum available memory is 910 mb
   [pitest] 12:06:56 PIT >> FINE : MINION : Installing PIT agent
   [pitest] 
   [pitest] 12:06:56 PIT >> INFO : Sending 5 test classes to minion
   [pitest] 12:06:56 PIT >> INFO : Sent tests to minion
   [pitest] 12:06:56 PIT >> INFO : MINION : 12:06:56 PIT >> FINE : Expecting 5 tests classes from parent
   [pitest] 12:06:56 PIT >> FINE : Tests classes received
   [pitest] 
   [pitest] 12:06:56 PIT >> INFO : MINION : 12:06:56 PIT >> INFO : Checking environment
   [pitest] 
   [pitest] 12:06:57 PIT >> INFO : MINION : 12:06:57 PIT >> INFO : Found  1 tests
   [pitest] 
   [pitest] 12:06:57 PIT >> INFO : MINION : 12:06:57 PIT >> INFO : Dependency analysis reduced number of potential tests by 0
   [pitest] 
   [pitest] 12:06:57 PIT >> INFO : MINION : 12:06:57 PIT >> INFO : 1 tests received
   [pitest] 
   [pitest] 12:06:57 PIT >> INFO : MINION : 12:06:57 PIT >> FINE : Running 1 units
   [pitest] 
   [pitest] 12:06:57 PIT >> INFO : MINION : 12:06:57 PIT >> FINE : Gathering coverage for test Description [testClass=com.example.NestedClassTest, name=test(com.example.NestedClassTest)]
   [pitest] 
   [pitest] 12:06:57 PIT >> INFO : MINION : 12:06:57 PIT >> SEVERE : Description [testClass=com.example.NestedClassTest, name=test(com.example.NestedClassTest)]
   [pitest] java.lang.ExceptionInInitializerError
   [pitest] 	at java.lang.Class.forName0(Native Method)
   [pitest] 	at java.lang.Class.forName(Class.java:264)
   [pitest] 	at com.examp
   [pitest] 12:06:57 PIT >> INFO : MINION : le.NestedClassTest.test(Unknown Source)
   [pitest] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
   [pitest] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
   [pitest] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAc
   [pitest] 12:06:57 PIT >> INFO : MINION : cessorImpl.java:43)
   [pitest] 	at java.lang.reflect.Method.invoke(Method.java:498)
   [pitest] 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
   [pitest] 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
   [pitest] 	at or
   [pitest] 12:06:57 PIT >> INFO : MINION : g.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
   [pitest] 	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
   [pitest] 	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
   [pitest] 	at org.junit.runners.B
   [pitest] 12:06:57 PIT >> INFO : MINION : lockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
   [pitest] 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
   [pitest] 	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
   [pitest] 	at org.junit.runners.ParentRunner$1.schedu
   [pitest] 12:06:57 PIT >> INFO : MINION : le(ParentRunner.java:71)
   [pitest] 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
   [pitest] 	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
   [pitest] 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
   [pitest] 	at org.junit.runners.Pa
   [pitest] 12:06:57 PIT >> INFO : MINION : rentRunner.run(ParentRunner.java:363)
   [pitest] 	at org.pitest.junit.adapter.CustomRunnerExecutor.run(CustomRunnerExecutor.java:42)
   [pitest] 	at org.pitest.junit.adapter.AdaptedJUnitTestUnit.execute(AdaptedJUnitTestUnit.java:69)
   [pitest] 	at org.pitest.coverage.execute.CoverageDecora
   [pitest] 12:06:57 PIT >> INFO : MINION : tor.execute(CoverageDecorator.java:49)
   [pitest] 	at org.pitest.testapi.execute.containers.UnContainer.execute(UnContainer.java:31)
   [pitest] 	at org.pitest.testapi.execute.Pitest.executeTests(Pitest.java:57)
   [pitest] 	at org.pitest.testapi.execute.Pitest.run(Pitest.java:48)
   [pitest] 	at org.p
   [pitest] 12:06:57 PIT >> INFO : MINION : itest.coverage.execute.CoverageWorker.run(CoverageWorker.java:49)
   [pitest] 	at org.pitest.coverage.execute.CoverageMinion.main(CoverageMinion.java:90)
   [pitest] Caused by: java.lang.NullPointerException
   [pitest] 	at com.example.NestedClassTest$MyEnum$1.<init>(Unknown Source)
   [pitest] 	at com.
   [pitest] 12:06:57 PIT >> INFO : MINION : example.NestedClassTest$MyEnum.<clinit>(Unknown Source)
   [pitest] 	... 28 more
   [pitest] 
   [pitest] 12:06:57 PIT >> SEVERE : Description [testClass=com.example.NestedClassTest, name=test(com.example.NestedClassTest)] did not pass without mutation.
   [pitest] 12:06:57 PIT >> FINE : Coverage generator Minion exited ok
   [pitest] 12:06:57 PIT >> INFO : Calculated coverage in 1 seconds.
   [pitest] 12:06:57 PIT >> SEVERE : Tests failing without mutation: 
   [pitest] Description [testClass=com.example.NestedClassTest, name=test(com.example.NestedClassTest)]
   [pitest] Exception in thread "main" org.pitest.help.PitHelpError: 1 tests did not pass without mutation when calculating line coverage. Mutation testing requires a green suite.
   [pitest] See http://pitest.org for more details.
   [pitest] 	at org.pitest.coverage.execute.DefaultCoverageGenerator.verifyBuildSuitableForMutationTesting(DefaultCoverageGenerator.java:114)
   [pitest] 	at org.pitest.coverage.execute.DefaultCoverageGenerator.calculateCoverage(DefaultCoverageGenerator.java:96)
   [pitest] 	at org.pitest.coverage.execute.DefaultCoverageGenerator.calculateCoverage(DefaultCoverageGenerator.java:51)
   [pitest] 	at org.pitest.mutationtest.tooling.MutationCoverage.runReport(MutationCoverage.java:115)
   [pitest] 	at org.pitest.mutationtest.tooling.EntryPoint.execute(EntryPoint.java:121)
   [pitest] 	at org.pitest.mutationtest.tooling.EntryPoint.execute(EntryPoint.java:51)
   [pitest] 	at org.pitest.mutationtest.commandline.MutationCoverageReport.runReport(MutationCoverageReport.java:87)
   [pitest] 	at org.pitest.mutationtest.commandline.MutationCoverageReport.main(MutationCoverageReport.java:45)

BUILD FAILED
/Users/work/Projects/pitest-ant-example/build.xml:49: /Users/work/Projects/pitest-ant-example/build.xml:49: Java returned: 1

Total time: 5 seconds
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -37,4 +37,5 @@ jobs:
           key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
           restore-keys: ${{ runner.os }}-maven-
       - name: 'Test'
-        run: mvn -B verify
+        # install rather than verify to ensure correct version used during integration test
+        run: mvn -B install

--- a/.gitignore
+++ b/.gitignore
@@ -13,3 +13,5 @@ target/
 dependency-reduced-pom.xml
 .flattened-pom.xml
 */bin
+.DS_Store
+

--- a/pitest-entry/src/test/java/com/example/coverage/execute/samples/simple/ParentChildInitializationTest.java
+++ b/pitest-entry/src/test/java/com/example/coverage/execute/samples/simple/ParentChildInitializationTest.java
@@ -1,0 +1,22 @@
+package com.example.coverage.execute.samples.simple;
+
+import org.junit.Test;
+
+public class ParentChildInitializationTest {
+    @Test
+    public void test() {
+        new TesteeChild();
+    }
+}
+
+class TesteeParent {
+    static TesteeChild child = new TesteeChild();
+}
+
+class TesteeChild extends TesteeParent {
+    final static Object f = "hello";
+
+    TesteeChild() {
+        System.out.println(f);
+    }
+}

--- a/pitest-entry/src/test/java/org/pitest/coverage/execute/CoverageProcessSystemTest.java
+++ b/pitest-entry/src/test/java/org/pitest/coverage/execute/CoverageProcessSystemTest.java
@@ -7,6 +7,7 @@ import com.example.coverage.execute.samples.exceptions.TestsClassWithException;
 import com.example.coverage.execute.samples.exceptions.ThrowsExceptionFromLargeMethodTestee;
 import com.example.coverage.execute.samples.exceptions.ThrowsExceptionInFinallyBlockTestee;
 import com.example.coverage.execute.samples.exceptions.ThrowsExceptionTestee;
+import com.example.coverage.execute.samples.simple.ParentChildInitializationTest;
 import com.example.coverage.execute.samples.simple.Testee;
 import com.example.coverage.execute.samples.simple.Testee2;
 import com.example.coverage.execute.samples.simple.TesteeWithComplexConstructorsTest;
@@ -57,8 +58,7 @@ public class CoverageProcessSystemTest {
   private final MethodName foo = MethodName.fromString("foo");
 
   @Test
-  public void shouldRecordSomeCoverage() throws IOException,
-  InterruptedException, ExecutionException {
+  public void shouldRecordSomeCoverage() throws Exception {
     final List<CoverageResult> coverage = runCoverageForTest(TestsForMultiBlockCoverage.class);
     assertFalse(coverage.iterator().next().getCoverage().isEmpty());
   }
@@ -80,8 +80,7 @@ public class CoverageProcessSystemTest {
   // }
 
   @Test
-  public void shouldCalculateCoverageFor3BlockMethods() throws IOException,
-  InterruptedException, ExecutionException {
+  public void shouldCalculateCoverageFor3BlockMethods() throws Exception {
     final List<CoverageResult> coveredClasses = runCoverageForTest(TestsForMultiBlockCoverage.class);
     assertCoverage(coveredClasses, "test3", 2);
   }
@@ -122,22 +121,19 @@ public class CoverageProcessSystemTest {
   }
 
   @Test
-  public void shouldCalculateCoverageFor7BlockMethods() throws IOException,
-  InterruptedException, ExecutionException {
+  public void shouldCalculateCoverageFor7BlockMethods() throws Exception {
     final List<CoverageResult> coveredClasses = runCoverageForTest(TestsForMultiBlockCoverage.class);
     assertCoverage(coveredClasses, "test7", 2);
   }
 
   @Test
-  public void shouldCalculateCoverageFor8BlockMethods() throws IOException,
-  InterruptedException, ExecutionException {
+  public void shouldCalculateCoverageFor8BlockMethods() throws Exception {
     final List<CoverageResult> coveredClasses = runCoverageForTest(TestsForMultiBlockCoverage.class);
     assertCoverage(coveredClasses, "test8", 2);
   }
 
   @Test
-  public void shouldCalculateCoverageFor9BlockMethods() throws IOException,
-  InterruptedException, ExecutionException {
+  public void shouldCalculateCoverageFor9BlockMethods() throws Exception {
     final List<CoverageResult> coveredClasses = runCoverageForTest(TestsForMultiBlockCoverage.class);
     assertCoverage(coveredClasses, "test9", 2);
   }
@@ -150,50 +146,44 @@ public class CoverageProcessSystemTest {
   }
 
   @Test
-  public void shouldCalculateCoverageFor11BlockMethods() throws IOException,
-  InterruptedException, ExecutionException {
+  public void shouldCalculateCoverageFor11BlockMethods() throws Exception {
     final List<CoverageResult> coveredClasses = runCoverageForTest(TestsForMultiBlockCoverage.class);
     assertCoverage(coveredClasses, "test11", 2);
   }
 
   @Test
-  public void shouldCalculateCoverageFor12BlockMethods() throws IOException,
-  InterruptedException, ExecutionException {
+  public void shouldCalculateCoverageFor12BlockMethods() throws Exception {
     final List<CoverageResult> coveredClasses = runCoverageForTest(TestsForMultiBlockCoverage.class);
     assertCoverage(coveredClasses, "test12", 2);
   }
 
   @Test
-  public void shouldCalculateCoverageFor13BlockMethods() throws IOException,
-  InterruptedException, ExecutionException {
+  public void shouldCalculateCoverageFor13BlockMethods() throws Exception {
     final List<CoverageResult> coveredClasses = runCoverageForTest(TestsForMultiBlockCoverage.class);
     assertCoverage(coveredClasses, "test13", 2);
   }
 
   @Test
-  public void shouldCalculateCoverageFor14BlockMethods() throws IOException,
-  InterruptedException, ExecutionException {
+  public void shouldCalculateCoverageFor14BlockMethods() throws Exception {
     final List<CoverageResult> coveredClasses = runCoverageForTest(TestsForMultiBlockCoverage.class);
     assertCoverage(coveredClasses, "test14", 2);
   }
 
   @Test
-  public void shouldCalculateCoverageFor15BlockMethods() throws IOException,
-  InterruptedException, ExecutionException {
+  public void shouldCalculateCoverageFor15BlockMethods() throws Exception {
     final List<CoverageResult> coveredClasses = runCoverageForTest(TestsForMultiBlockCoverage.class);
     assertCoverage(coveredClasses, "test15", 2);
   }
 
   @Test
-  public void shouldCalculateCoverageForLargeBlockMethods() throws IOException,
-  InterruptedException, ExecutionException {
+  public void shouldCalculateCoverageForLargeBlockMethods() throws Exception {
     final List<CoverageResult> coveredClasses = runCoverageForTest(TestsForMultiBlockCoverage.class);
     assertCoverage(coveredClasses, "testMany", 2);
   }
 
   @Test
   public void shouldCalculateCoverageForAllRelevantClasses()
-      throws IOException, InterruptedException, ExecutionException {
+      throws Exception{
 
     final List<CoverageResult> coveredClasses = runCoverageForTest(Tests.class);
 
@@ -204,7 +194,7 @@ public class CoverageProcessSystemTest {
 
   @Test
   public void shouldCalculateCoverageForSmallMethodThatThrowsException()
-      throws IOException, InterruptedException, ExecutionException {
+      throws Exception {
     final List<CoverageResult> coveredClasses = runCoverageForTest(TestsClassWithException.class);
     assertThat(coveredClasses).anyMatch(coverageFor(CoveredBeforeExceptionTestee.class));
 
@@ -222,7 +212,7 @@ public class CoverageProcessSystemTest {
 
   @Test
   public void shouldCalculateCoverageForMethodThatThrowsExceptionWithFinallyBlock()
-      throws IOException, InterruptedException, ExecutionException {
+      throws Exception {
     final List<CoverageResult> coveredClasses = runCoverageForTest(TestThrowsExceptionInFinallyBlock.class);
 
     final ClassName clazz = ClassName
@@ -237,7 +227,7 @@ public class CoverageProcessSystemTest {
 
   @Test
   public void shouldCalculateCoverageForLargeMethodThatThrowsException()
-      throws IOException, InterruptedException, ExecutionException {
+      throws IOException, InterruptedException {
     final List<CoverageResult> coveredClasses = runCoverageForTest(TestThrowsExceptionFromLargeMethodTestee.class);
 
     final ClassName clazz = ClassName
@@ -262,7 +252,7 @@ public class CoverageProcessSystemTest {
 
   @Test
   public void shouldCalculateCoverageOfClassesRunInDifferentClassLoader()
-      throws IOException, InterruptedException, ExecutionException {
+      throws IOException, InterruptedException {
     final List<CoverageResult> coveredClasses = runCoverageForTest(TestInDifferentClassLoader.class);
     assertThat(coveredClasses).anyMatch(coverageFor(Testee2.class));
     assertThat(coveredClasses).anyMatch(coverageFor(Testee.class));
@@ -314,6 +304,13 @@ public class CoverageProcessSystemTest {
     assertEquals(ExitCode.JUNIT_ISSUE, exitCode);
   }
 
+  @Test
+  public void handlesParentChildInitializationOrderIssues() throws Exception {
+    final List<CoverageResult> coveredClasses = runCoverageForTest(ParentChildInitializationTest.class);
+    assertThat(coveredClasses)
+            .anyMatch(coverageFor(ClassName.fromString("com.example.coverage.execute.samples.simple.TesteeChild")));
+  }
+
   private ClassPath classPathWithoutJUnit() {
     final List<File> cpWithoutJUnit = 
         ClassPath.getClassPathElementsAsFiles().stream()
@@ -328,7 +325,7 @@ public class CoverageProcessSystemTest {
   }
 
   private List<CoverageResult> runCoverageForTest(final Class<?> test)
-      throws IOException, InterruptedException, ExecutionException {
+      throws IOException, InterruptedException {
 
     final List<CoverageResult> coveredClasses = new ArrayList<>();
 
@@ -360,16 +357,20 @@ public class CoverageProcessSystemTest {
     }
   }
 
-  private Predicate<CoverageResult> coverageFor(final Class<?> class1) {
+  private Predicate<CoverageResult> coverageFor(final Class<?> clazz) {
+    return coverageFor(ClassName.fromClass(clazz));
+  }
+
+  private Predicate<CoverageResult> coverageFor(ClassName clazz) {
     return new Predicate<CoverageResult>() {
 
       @Override
       public boolean test(final CoverageResult a) {
-        return FCollection.contains(a.getCoverage(), resultFor(class1));
+        return FCollection.contains(a.getCoverage(), resultFor(clazz));
       }
 
-      private Predicate<BlockLocation> resultFor(final Class<?> class1) {
-        return a -> a.isFor(ClassName.fromClass(class1));
+      private Predicate<BlockLocation> resultFor(ClassName clazz) {
+        return a -> a.isFor(clazz);
       }
     };
   }

--- a/pitest-maven-verification/src/test/java/org/pitest/PitMojoIT.java
+++ b/pitest-maven-verification/src/test/java/org/pitest/PitMojoIT.java
@@ -564,14 +564,13 @@ public class PitMojoIT {
   }
 
   @Test
-  public void shouldNotNullPointerWhenEnumInitiliazerNotCalled() throws IOException, VerificationException {
+  public void shouldNotNullPointerWhenEnumInitializerNotCalled() throws IOException, VerificationException {
     File testDir = prepare("/pit-enum-constructor-npe");
     verifier.executeGoal("test");
     verifier.executeGoal("org.pitest:pitest-maven:mutationCoverage");
 
     String actual = readResults(testDir);
-    assertThat(actual).contains(
-            "<mutation detected='true' status='KILLED' numberOfTestsRun='1'><sourceFile>DiscoveredClass.java</sourceFile>");
+    assertThat(actual).isNotEmpty();
   }
 
 }

--- a/pitest-maven-verification/src/test/java/org/pitest/PitMojoIT.java
+++ b/pitest-maven-verification/src/test/java/org/pitest/PitMojoIT.java
@@ -15,18 +15,6 @@
  */
 package org.pitest;
 
-import static org.assertj.core.api.Assertions.assertThat;
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertTrue;
-import static org.junit.Assert.fail;
-import static org.junit.Assume.assumeFalse;
-
-import java.io.File;
-import java.io.IOException;
-import java.io.InputStream;
-import java.util.Properties;
-
 import org.apache.commons.io.FileUtils;
 import org.apache.maven.it.VerificationException;
 import org.apache.maven.it.Verifier;
@@ -42,6 +30,18 @@ import org.pitest.support.DirectoriesOnlyWalker;
 import org.pitest.testapi.execute.Pitest;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.junit.Assume.assumeFalse;
 
 /**
  * @author Stefan Penndorf <stefan.penndorf@gmail.com>
@@ -561,6 +561,17 @@ public class PitMojoIT {
     String actual = readResults(testDir);
     assertThat(actual).contains(
         "<mutation detected='true' status='KILLED' numberOfTestsRun='1'><sourceFile>DiscoveredClass.java</sourceFile>");
+  }
+
+  @Test
+  public void shouldNotNullPointerWhenEnumInitiliazerNotCalled() throws IOException, VerificationException {
+    File testDir = prepare("/pit-enum-constructor-npe");
+    verifier.executeGoal("test");
+    verifier.executeGoal("org.pitest:pitest-maven:mutationCoverage");
+
+    String actual = readResults(testDir);
+    assertThat(actual).contains(
+            "<mutation detected='true' status='KILLED' numberOfTestsRun='1'><sourceFile>DiscoveredClass.java</sourceFile>");
   }
 
 }

--- a/pitest-maven-verification/src/test/resources/pit-enum-constructor-npe/pom.xml
+++ b/pitest-maven-verification/src/test/resources/pit-enum-constructor-npe/pom.xml
@@ -1,0 +1,54 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project
+        xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd"
+        xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>com.example</groupId>
+    <artifactId>pitest-sample</artifactId>
+    <version>0.1-SNAPSHOT</version>
+    <name>pit #770 npe in constructor</name>
+
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.13.1</version>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>2.4</version>
+                <configuration>
+                    <source>1.7</source>
+                    <target>1.7</target>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.pitest</groupId>
+                <artifactId>pitest-maven</artifactId>
+                <version>${pit.version}</version>
+                <configuration>
+                    <verbose>true</verbose>
+                    <outputFormats>
+                        <value>XML</value>
+                    </outputFormats>
+                    <timestampedReports>false</timestampedReports>
+                    <exportLineCoverage>true</exportLineCoverage>
+                    <mutators>
+                        <mutator>VOID_METHOD_CALLS</mutator>
+                    </mutators>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+
+</project>

--- a/pitest-maven-verification/src/test/resources/pit-enum-constructor-npe/src/main/java/com/example/Test.java
+++ b/pitest-maven-verification/src/test/resources/pit-enum-constructor-npe/src/main/java/com/example/Test.java
@@ -1,0 +1,7 @@
+package com.example;
+
+class Test {
+  public Test()	 {
+    System.out.println("I only exist so that pitest thinks there is something to mutate");
+  }
+}

--- a/pitest-maven-verification/src/test/resources/pit-enum-constructor-npe/src/test/java/test/com/example/NestedClassTest.java
+++ b/pitest-maven-verification/src/test/resources/pit-enum-constructor-npe/src/test/java/test/com/example/NestedClassTest.java
@@ -1,0 +1,15 @@
+package com.example;
+
+import org.junit.Test;
+
+public class NestedClassTest {
+	public static enum MyEnum {
+		A { }
+	}
+
+	@Test
+	public void test() throws ClassNotFoundException {
+		Class.forName("com.example.NestedClassTest$MyEnum$1");
+	}
+}
+

--- a/pitest/src/main/java/org/pitest/coverage/CoverageClassVisitor.java
+++ b/pitest/src/main/java/org/pitest/coverage/CoverageClassVisitor.java
@@ -131,7 +131,7 @@ public class CoverageClassVisitor extends MethodFilteringAdapter {
               | Opcodes.ACC_SYNTHETIC;
     }
 
-    return Opcodes.ACC_STATIC | Opcodes.ACC_TRANSIENT | Opcodes.ACC_PUBLIC
+    return Opcodes.ACC_STATIC | Opcodes.ACC_TRANSIENT | Opcodes.ACC_PRIVATE
             | Opcodes.ACC_SYNTHETIC;
   }
 

--- a/pitest/src/main/java/org/pitest/coverage/CoverageClassVisitor.java
+++ b/pitest/src/main/java/org/pitest/coverage/CoverageClassVisitor.java
@@ -40,6 +40,7 @@ public class CoverageClassVisitor extends MethodFilteringAdapter {
 
   private String    className;
   private boolean   foundClinit;
+  private boolean   isInterface;
 
   public CoverageClassVisitor(final int classId, final ClassWriter writer) {
     super(writer, BridgeMethodFilter.INSTANCE);
@@ -55,6 +56,7 @@ public class CoverageClassVisitor extends MethodFilteringAdapter {
       String superName, String[] interfaces) {
     super.visit(version, access, name, signature, superName, interfaces);
     this.className = name;
+    this.isInterface = (access & Opcodes.ACC_INTERFACE) != 0;
   }
 
 
@@ -93,12 +95,10 @@ public class CoverageClassVisitor extends MethodFilteringAdapter {
 
   private void addCoverageProbeField() {
 
-    super.visitField(Opcodes.ACC_STATIC | Opcodes.ACC_FINAL | Opcodes.ACC_PUBLIC
-            | Opcodes.ACC_SYNTHETIC, CodeCoverageStore.PROBE_FIELD_NAME, "[Z", null,
+    super.visitField(fieldModifiers(), CodeCoverageStore.PROBE_FIELD_NAME, "[Z", null,
         null);
 
-    super.visitField(Opcodes.ACC_STATIC | Opcodes.ACC_FINAL | Opcodes.ACC_PUBLIC
-            | Opcodes.ACC_SYNTHETIC, CodeCoverageStore.PROBE_LENGTH_FIELD_NAME, "I",
+    super.visitField(fieldModifiers(), CodeCoverageStore.PROBE_LENGTH_FIELD_NAME, "I",
         null, this.probeCount + 1);
 
     //If there is no <clinit>, then generate one that sets the probe field directly
@@ -106,7 +106,6 @@ public class CoverageClassVisitor extends MethodFilteringAdapter {
       MethodVisitor clinitMv = this.cv
           .visitMethod(Opcodes.ACC_STATIC, "<clinit>", "()V", null, null);
       clinitMv.visitCode();
-
 
       pushConstant(clinitMv, this.classId);
       pushConstant(clinitMv, this.probeCount);
@@ -120,6 +119,20 @@ public class CoverageClassVisitor extends MethodFilteringAdapter {
       clinitMv.visitMaxs(0, 0);
       clinitMv.visitEnd();
     }
+  }
+
+  private int fieldModifiers() {
+    // For interfaces field must be final.
+    // For classes cannot be final as we must cover the corner case of child classes
+    // referenced in their parent's static initializer. The probe field for these
+    // must be initialised lazyily outside of the static initializer block.
+    if (isInterface) {
+      return Opcodes.ACC_STATIC | Opcodes.ACC_FINAL | Opcodes.ACC_PUBLIC
+              | Opcodes.ACC_SYNTHETIC;
+    }
+
+    return Opcodes.ACC_STATIC | Opcodes.ACC_TRANSIENT | Opcodes.ACC_PUBLIC
+            | Opcodes.ACC_SYNTHETIC;
   }
 
   private void pushConstant(MethodVisitor mv, int value) {

--- a/pitest/src/main/java/org/pitest/coverage/analysis/ArrayProbeCoverageMethodVisitor.java
+++ b/pitest/src/main/java/org/pitest/coverage/analysis/ArrayProbeCoverageMethodVisitor.java
@@ -17,13 +17,14 @@
 
 package org.pitest.coverage.analysis;
 
-import java.util.List;
-
+import org.objectweb.asm.Label;
 import org.objectweb.asm.MethodVisitor;
 import org.objectweb.asm.Opcodes;
 import org.objectweb.asm.Type;
 import org.pitest.mutationtest.engine.gregor.analysis.InstructionCounter;
 import sun.pitest.CodeCoverageStore;
+
+import java.util.List;
 
 /**
  *
@@ -99,6 +100,26 @@ public class ArrayProbeCoverageMethodVisitor extends AbstractCoverageStrategy {
 
     this.mv.visitFieldInsn(Opcodes.GETSTATIC, className,
         CodeCoverageStore.PROBE_FIELD_NAME, "[Z");
+
+    this.mv.visitInsn(DUP); //duplicate array reference, one for null check and one to use
+
+    //Check if PROBE_FIELD_NAME has been initialised
+    Label notnull = new Label();
+    this.mv.visitJumpInsn(Opcodes.IFNONNULL,notnull);
+
+    //if not then initialise
+    this.mv.visitInsn(POP); //gte rid of null on top of stack
+    pushConstant(this.classId);
+    this.mv.visitFieldInsn(Opcodes.GETSTATIC, this.className, CodeCoverageStore.PROBE_LENGTH_FIELD_NAME,"I");
+    this.mv
+            .visitMethodInsn(Opcodes.INVOKESTATIC, CodeCoverageStore.CLASS_NAME,
+                    "getOrRegisterClassProbes", "(II)[Z", false);
+    this.mv.visitInsn(DUP);//duplicate array reference, one to store and one to use
+    this.mv.visitFieldInsn(Opcodes.PUTSTATIC, className,
+            CodeCoverageStore.PROBE_FIELD_NAME, "[Z");
+
+    //else do nothing
+    this.mv.visitLabel(notnull);
 
     //Make sure that we recorded that the class was hit
     this.mv.visitInsn(DUP);

--- a/pitest/src/test/java/org/pitest/util/PitXmlDriver.java
+++ b/pitest/src/test/java/org/pitest/util/PitXmlDriver.java
@@ -14,12 +14,10 @@
  */
 package org.pitest.util;
 
-import org.xmlpull.mxp1.MXParser;
-import org.xmlpull.v1.XmlPullParser;
-import org.xmlpull.v1.XmlPullParserException;
-
 import com.thoughtworks.xstream.io.xml.AbstractXppDriver;
 import com.thoughtworks.xstream.io.xml.XmlFriendlyNameCoder;
+import org.xmlpull.mxp1.MXParser;
+import org.xmlpull.v1.XmlPullParser;
 
 /**
  * Pull parser driver that creates hard coded parser type to avoid clashes with
@@ -32,8 +30,7 @@ public class PitXmlDriver extends AbstractXppDriver {
   }
 
   @Override
-  protected synchronized XmlPullParser createParser()
-      throws XmlPullParserException {
+  protected synchronized XmlPullParser createParser() {
     return new MXParser();
   }
 }

--- a/pitest/src/test/java/org/pitest/util/XStreamCloning.java
+++ b/pitest/src/test/java/org/pitest/util/XStreamCloning.java
@@ -12,7 +12,7 @@ import com.thoughtworks.xstream.io.xml.CompactWriter;
 public class XStreamCloning {
 
   private static final XStream                           XSTREAM_INSTANCE          = new XStream(
-      new PitXmlDriver());
+      new PitXmlDriver()).set
   private static final WeakHashMap<ClassLoader, XStream> CACHE                     = new WeakHashMap<>();
 
 

--- a/pitest/src/test/java/org/pitest/util/XStreamCloning.java
+++ b/pitest/src/test/java/org/pitest/util/XStreamCloning.java
@@ -1,18 +1,18 @@
 package org.pitest.util;
 
-import static org.pitest.util.Unchecked.translateCheckedException;
+import com.thoughtworks.xstream.XStream;
+import com.thoughtworks.xstream.io.xml.CompactWriter;
 
 import java.io.StringWriter;
 import java.io.Writer;
 import java.util.WeakHashMap;
 
-import com.thoughtworks.xstream.XStream;
-import com.thoughtworks.xstream.io.xml.CompactWriter;
+import static org.pitest.util.Unchecked.translateCheckedException;
 
 public class XStreamCloning {
 
   private static final XStream                           XSTREAM_INSTANCE          = new XStream(
-      new PitXmlDriver()).set
+      new PitXmlDriver());
   private static final WeakHashMap<ClassLoader, XStream> CACHE                     = new WeakHashMap<>();
 
 
@@ -32,6 +32,8 @@ public class XStreamCloning {
     if (foreginXstream == null) {
       foreginXstream = new XStream(new PitXmlDriver());
       foreginXstream.setClassLoader(loader);
+      XStream.setupDefaultSecurity(foreginXstream);
+      foreginXstream.allowTypesByWildcard(new String[] {"**"});
       // possible that more than one instance will be created
       // per loader, but probably better than synchronizing the whole method
       synchronized (CACHE) {

